ARCs/arc-draft_extended_desc.md

```diff
@@ -0,0 +1,160 @@
+---
+arc: <to be assigned>
+title: Extended App Description
+description: Adds more information to the ARC4 JSON description
+author: Joe Polny (@joe-p)
+discussions-to: <URL>
+status: Draft
+type: Standards Track
+category: ARC
+created: 2023-11-14
+requires: 4
+---
+
+
+## Abstract
+This ARC takes the existing JSON description of a contract as described in ARC4 and adds more fields for the purpose of client interaction
+
+## Motivation
+The description provided by ARC4 is missing a lot of critical information that clients should know when interaction with an app.
+
+## Specification
+```ts
+/** Mapping of named structs to the ABI type of their fields */
+interface StructFields {
```

Comment:
This one is an associative array, but all of the other things seem to be arrays with objects that have a name property. Should we be consistent?

Reply:
I'm not sure I follow, what do you mean?

Reply:
i.e. this one is:

```
{
   fieldName: ...
}
```

rather than the others parts of the spec that are:

```
[{name: "fieldName", ...}]
```